### ISSUE TITLE
AddVariableDeclarationsStage failed to parse: Unexpected token due to parens parsing bug

### ISSUE THREAD

**Issue by avonwyss** [decaffeinate/decaffeinate#315]
([repl](http://decaffeinate-project.org/repl/#?evaluate=true&code=if%20a%20%26%26%20%28b%3D%3D'%29'%29%0A%20%20x))
```
if a && (b==')')
  x
```
The `)` enclosed in the string seems to be counted towards the total number of closing parens, so that the `{` is inserted at the wrong location:
```
AddVariableDeclarationsStage failed to parse: Unexpected token (1:19)
> 1 | if (a && (b===')') {)
    |                    ^
  2 |   x;
  3 | }
```

**Comment by alangpierce** [comment]
Dupe of #270. This is fixed on master in decaffeinate-parser, but the fix hasn't been released yet. @eventualbuddha, could you do another release for decaffeinate-parser?

**Comment by eventualbuddha** [comment]
[Fixed in v2.17.2](http://decaffeinate-project.org/repl/#?evaluate=false&code=if%20a%20%26%26%20(b%3D%3D')')%0A%20%20x).
